### PR TITLE
versatile-data-kit: copyright notice year update

### PR DESCRIPTION
**Pull request by ivakoleva**
versatile-data-kit: copyright notice year update

Copyright notice year is out of date.

Introduced a year range from 2021 to now.
Cleared up any messy leftovers.
Reconfigured the pre-commit hook to handle the year flexibly:
https://github.com/Lucas-C/pre-commit-hooks/blob/master/README.md#handling-years-flexibly

Testing done: pre-commit hooks triggerred

Signed-off-by: ivakoleva <iva.koleva@clearcode.bg>

### CODE REVIEW

**Comment by ivakoleva** [comment]
pre-commit.ci autofix

**Review by duyguHsnHsn — APPROVED** [review]

**Review by DeltaMichael — CHANGES_REQUESTED** [review]
I'll approve it, but can you please open a task to automate updating the headers with the current year at the start of each year? (or to check if that's possible).

File: cicd/build.sh [hunk 1]
```diff
@@ -1,4 +1,7 @@
 #!/bin/bash -e
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
 # Copyright 2021 VMware, Inc.
```
Comment (ivakoleva) [inline cicd/build.sh:5]: You've probably noticed the bot is prepending the new notice directly-
am now cleaning up the double and sometimes even triple notices (leftovers from prev notice format update).

Comment (DeltaMichael) [reply cicd/build.sh:5]: Changing one digit is easy, I agree, it's just the part we have to remember to do it every year that bugs me. Anyway, I think you've discovered a much worse problem that deserves a ticket.

Comment (ivakoleva) [reply cicd/build.sh:5]: another leftover variation would be:
`# Copyright 2021 VMware, Inc.
# SPDX-License-Identifier: Apache-2.0
# Copyright 2022 VMware, Inc.
# SPDX-License-Identifier: Apache-2.0
`

Comment (ivakoleva) [reply cicd/build.sh:5]: I also noticed the .yaml & .yml files are not configured for handling,
yet they (at least some) had old notices added. I will be addressing that in the pre-commit hook reconfigured PR https://github.com/vmware/versatile-data-kit/pull/1625

**Review by DeltaMichael — APPROVED** [review]

**Pull request merged**